sys/drv/pit.c

```diff
@@ -115,24 +115,25 @@ static int timer_pit_start(timer_t *tm, unsigned flags, const bintime_t start,
   return 0;
 }
 
-static int timer_pit_stop(timer_t *tm) {
+static int pit_timer_stop(timer_t *tm) {
   device_t *dev = device_of(tm);
   pit_state_t *pit = dev->state;
   bus_intr_teardown(dev, pit->irq_res);
   return 0;
 }
 
-static bintime_t timer_pit_gettime(timer_t *tm) {
+static bintime_t pit_timer_gettime(timer_t *tm) {
   device_t *dev = device_of(tm);
   pit_state_t *pit = dev->state;
-  uint32_t sec, ticks;
+  uint64_t sec, ticks;
   WITH_INTR_DISABLED {
     pit_update_time(pit);
     sec = pit->sec;
     ticks = pit->ticks;
   }
   bintime_t bt = bintime_mul(tm->tm_min_period, ticks);
-  bt.sec += sec;
+  assert(bt.sec == 0);
```

Comment:
Why such an assumption?

Reply:
Cause the multiplied value is a modulo of timer frequency and the sum of time which the ticks took should be less than a second.